pkg/sdk/metrics/metrics.go

```diff
@@ -0,0 +1,74 @@
+package metrics
+
+import (
+	"sync"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	eventsMetricName     = "operator_sdk_event_type"
```

Comment:
Just noticed the `operator_sdk` prefix. I don't think that adds any more meaning to the metric name so we should probably drop it. Same for `operator_sdk_reconcile_result`.

A more meaningful prefix might be the controller name itself. But since we just have one collector per operator and until we support multiple controllers via the controller-runtime we can just keep it as `event_type` and `reconcile_result`.

Reply:
I think the metric name should also be shorter but i think having the word `operator` in there is useful as there might be plenty of non-operator metrics that have the string "event_type" in it. How about `operator_reconcile_results` and `operator_event_types`. Also you should use pluralization, and probably add a suffix of the unit for these metrics (like `_total`) see: https://prometheus.io/docs/practices/naming/ for best practices on naming.